File: go/LeetCode/InsertSort.go
package main

func main() {
	nums := []int{5,1,3,8,2}
	insertSort(nums)
	print(nums)
}

func insertSort(nums []int)  {
	len := len(nums)
	for i:=1;i<len;i++ {
		for j:=i;j>0 && nums[j]< nums[j-1];j-- {
			swap2(nums, j, j-1)
		}
	}
}

func swap2(nums []int, i, j int)  {
	temp := nums[i]
	nums[i] = nums[j]
	nums[j] = temp
}

File: python3/LeetCode/InsertSort.py
from typing import List


class Solution:
    def insertSort(self, nums:List[int]):
        length = len(nums)
        for i in range(1, length):
            for j in range(i, 0,-1):
                if nums[j]<nums[j-1]:
                    self.swap(nums, j, j-1)

    def swap(self, nums:List[int], i,j:int):
        temp = nums[i]
        nums[i] = nums[j]
        nums[j] = temp


if __name__ == '__main__':
    nums = [6,1,4,9,7,2]
    Solution().insertSort(nums)
    print(nums)